### PR TITLE
TEST: improves `tests\explainers\test_permutation.py`

### PR DESCRIPTION
**Pull request by naxatra2**
## Overview

Part of https://github.com/shap/shap/issues/3690

I have added tests in `shap\tests\explainers\test_permutation.py`. 
Description: 

- Refactored PermutationExplainer test suite to reduce redundancy using parametrization.
- Added new unit tests for:

  - max_evals lower bound validation
  - seed-based reproducibility
  - main_effects and error_bounds execution paths
  - constant feature edge case
- Improved overall test coverage and maintainability.

## Checklist

- [ x] All [pre-commit checks](https://pre-commit.com/#install) pass.
- [ x] Unit tests added (if fixing a bug or adding a new feature)

### CODE REVIEW

File: tests/explainers/test_permutation.py [hunk 1]
```diff
@@ -102,3 +100,53 @@ def test_serialization_custom_model_save():
...
+    data = np.random.rand(10, 3)
+
+    explainer = shap.explainers.PermutationExplainer(simple_sum_model, data)
+
+    with pytest.raises(ValueError):
+        explainer(data, max_evals=1)
+
+
+def test_seed_reproducibility():
+    data = np.random.rand(10, 4)
```
Comment (CloseChoice) [inline tests/explainers/test_permutation.py:115]: can you pls fix a seed for this?

File: tests/explainers/test_permutation.py [hunk 2]
```diff
@@ -102,3 +100,53 @@ def test_serialization_custom_model_save():
...
+    explainer2 = shap.explainers.PermutationExplainer(simple_sum_model, data, seed=42)
+
+    vals1 = explainer1(data).values
+    vals2 = explainer2(data).values
+
+    assert np.allclose(vals1, vals2)
+
+
+def test_main_effects():
+    data = np.random.rand(5, 3)
```
Comment (CloseChoice) [inline tests/explainers/test_permutation.py:127]: same here

File: tests/explainers/test_permutation.py [hunk 3]
```diff
@@ -102,3 +100,53 @@ def test_serialization_custom_model_save():
...
+    explainer = shap.explainers.PermutationExplainer(simple_sum_model, data)
+    explanation = explainer(data, main_effects=True)
+
+    # Just verify no crash and shape consistency if present
+    if explanation.main_effects is not None:
+        assert explanation.main_effects.shape == data.shape
+
+
+def test_error_bounds():
+    data = np.random.rand(5, 3)
```
Comment (CloseChoice) [inline tests/explainers/test_permutation.py:138]: same here

File: tests/explainers/test_permutation.py [hunk 4]
```diff
@@ -102,3 +100,53 @@ def test_serialization_custom_model_save():
...
+
+    explainer = shap.explainers.PermutationExplainer(simple_sum_model, data)
+    explanation = explainer(data, error_bounds=True)
+
+    assert explanation.error_std is not None
+
+
+def test_constant_features():
+    data = np.ones((10, 4))  # no variation
```
Comment (CloseChoice) [inline tests/explainers/test_permutation.py:147]: same

**Review by CloseChoice — CHANGES_REQUESTED** [review]
looks good, but pls fix the seeds

**Comment by naxatra2** [comment]
I have resolved the seed issues now. @CloseChoice
